docs/book/reflection-abstract-factory.md

```diff
@@ -2,14 +2,33 @@
 
 - Since 3.2.0.
 
-Writing a factory class for each and every service that has dependencies
-can be tedious, particularly in early development as you are still sorting
-out dependencies.
+Writing a factory class for each and every service that has dependencies can be tedious, particularly in early development as you are still sorting out dependencies.
 
-laminas-servicemanager ships with `Laminas\ServiceManager\AbstractFactory\ReflectionBasedAbstractFactory`,
-which provides a reflection-based approach to instantiation, resolving
-constructor dependencies to the relevant services. The factory may be used as
-either an abstract factory, or mapped to specific service names as a factory:
+To alleviate this issue during development, laminas-servicemanager ships with the `ReflectionBasedAbstractFactory`, which provides a [reflection-based approach](https://www.php.net/manual/en/intro.reflection.php) to instantiation, resolving constructor dependencies to the relevant services. 
+The factory may be used as either an abstract factory or mapped to specific service names as a factory.
+
+TIP: Mapping services to the factory is more explicit and performant.
+
+The factory operates with the following constraints/features:
+
+- A parameter named `$config` type-hinted as an array will receive the application "config" service (i.e., the merged configuration).
+- Parameters type-hinted against array, but not named `$config`, will be injected with an empty array.
+- Scalar parameters will result in the factory raising an exception, unless a default value is present; if it is, that value will be used.
+- If a service cannot be found for a given typehint, the factory will raise an exception detailing this.
+
+WARNING: `$options` passed to the factory are ignored in all cases, as we cannot make assumptions about which argument(s) they might replace.
+
+Once your dependencies have stabilized and when performance is a requirement, we recommend writing a dedicated factory, as reflection can introduce performance overhead. 
+For example, you could use the [generate-factory-for-class console tool](console-tools.md#generate-factory-for-class) to do so.
+
+## Usage Example
+
+### Using Laminas MVC
+
+When using [Laminas MVC](https://docs.laminas.dev/mvc/):
+
+- Add the factory in the service manager's list of abstract factories.
+- Use the `ReflectionBasedAbstractFactory` to instantiate the class.
```

Comment:
This part and also the code example are wrong because you do not need both options to use `ReflectionBasedAbstractFactory`.

- If the `ReflectionBasedAbstractFactory` is used an abstract factory, then it will do the job for all services to which no explicit factory is mapped.
- If the `ReflectionBasedAbstractFactory` is used as factory for a single service, then it will work only for this one service.

This means that we need two independent examples for the documentation, each with its own explanation!